tests/functional/testTutorials.js

```diff
@@ -52,28 +42,30 @@ describe('Tutorials', function () {
   describe('"Leave tutorial?" popup', function() {
     beforeEach(async () => {
       await openHighlightedTutorial();
-      await vpn.wait();
       await clickTooltipCloseButton();
-      await vpn.wait();
     });
 
     it('Clicking primary button closes modal and resumes tutorial',
-      async () => {
-        await vpn.waitForElementProperty(homeScreen.TUTORIAL_POPUP_PRIMARY_BUTTON, 'visible', 'true');
-        await vpn.clickOnElement(homeScreen.TUTORIAL_POPUP_PRIMARY_BUTTON);
-        await vpn.wait();
+       async () => {
+         await vpn.waitForElementAndClick(
+             homeScreen.TUTORIAL_POPUP_PRIMARY_BUTTON);
 
-        assert((await vpn.getElementProperty(homeScreen.TUTORIAL_POPUP_PRIMARY_BUTTON, 'visible')) === 'false');
-        assert((await vpn.getElementProperty(homeScreen.TUTORIAL_UI, 'visible')) === 'true');
-      });
+         assert(
+             (await vpn.getElementProperty(
+                 homeScreen.TUTORIAL_POPUP_PRIMARY_BUTTON, 'visible')) ===
+             'false');
+         assert(
+             (await vpn.getElementProperty(
+                 homeScreen.TUTORIAL_UI, 'visible')) === 'true');
+       });
 
     it('Clicking secondary button closes modal and stops tutorial',
-      async () => {
-        await vpn.waitForElementProperty(homeScreen.TUTORIAL_POPUP_SECONDARY_BUTTON, 'visible', 'true');
-        await vpn.clickOnElement(homeScreen.TUTORIAL_POPUP_SECONDARY_BUTTON);
-        await vpn.wait();
-
-        assert((await vpn.getElementProperty(homeScreen.TUTORIAL_UI, 'visible')) === 'false');
-      });
+       async () => {
+         await vpn.waitForElementAndClick(
+             homeScreen.TUTORIAL_POPUP_SECONDARY_BUTTON);
+         assert(
```

Comment:
i wonder if this would cause flakiness depending on how long it takes the modal to close?